File: 프로그래머스/2/42586. 기능개발/기능개발.py
import sys
import math
input = sys.stdin.readline


def solution(progresses, speeds):
    answer = []
    delay_list = [math.ceil((100-progresses[i])/speeds[i])
                  for i in range(len(progresses))]
    cnt = 1
    tmp = delay_list[0]
    for i in range(1, len(delay_list)):
        if tmp >= delay_list[i]:
            cnt += 1
        else:
            answer.append(cnt)
            tmp = delay_list[i]
            cnt = 1
    answer.append(cnt)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] arr = new int[progresses.length];
        for (int i=0; i<progresses.length; i++) {
            arr[i] = (int) Math.ceil((100 - progresses[i]) / (double) speeds[i]);
        }
        List<Integer> result = new ArrayList<>();
        int v = arr[0];
        int sum = 1;

        for (int i=1; i<arr.length; i++) {
            if (v >= arr[i]) {
                sum += 1;
            } else {
                result.add(sum);
                v = arr[i];
                sum = 1;
            }
        }

        result.add(sum);

        return result.stream().mapToInt(i -> i).toArray();
    }
}
